parachain/primitives/core/src/types.rs

```diff
@@ -3,20 +3,14 @@
 use codec::{Decode, Encode};
 use enum_iterator::IntoEnumIterator;
 use frame_support::{scale_info::TypeInfo, RuntimeDebug};
-use sp_core::H256;
+use sp_core::{H160, H256};
 use sp_runtime::DigestItem;
 use sp_std::vec::Vec;
 
 #[derive(Encode, Decode, Copy, Clone, PartialEq, Eq, RuntimeDebug, TypeInfo)]
-pub struct MessageId {
-	pub channel_id: ChannelId,
-	pub nonce: u64,
-}
-
-impl MessageId {
-	pub fn new(channel_id: ChannelId, nonce: u64) -> Self {
-		Self { channel_id, nonce }
-	}
+pub enum MessageId {
+	Basic(ChannelId, H160, u64),
```

Comment:
@vgeddes I'm not sure about adding the origin address to `MessageId` here, vs adding an extra id field to the message event from the outbound contract like we discussed.
Looks like the event's id field will just be the concatenation of the origin address and the nonce. And we need the separate origin address and nonce in the contract's event for the `Nonces` storage map anyway. Does it make more sense to do this concatenation only on the parachain, or to add a separate id field to the `Message` event in the contract?

Reply:
Yeah I would not add a redundant ID field to the `Message`, since we can generate the full ID on the parachain using the various fields in Message.

To add more background info, this `MessageId` type isn't really used in the core bridging protocol.  When a message is dispatched on the parachain, a `MessageDispatched` event is emitted [here](https://github.com/Snowfork/snowbridge/blob/820ec5d8d8e1a9973790e05ccc9f7d3ca88fac53/parachain/pallets/dispatch/src/lib.rs#L125), with the `MessageId` as a member. Offchain clients query these events to get progress updates on bridge transfers, and so the ID is necessary to correlate events happening on either chain.